raiden/network/transport/matrix/transport.py

```diff
@@ -459,19 +492,27 @@ def stop(self):
         # parent may want to call get() after stop(), to ensure _run errors are re-raised
         # we don't call it here to avoid deadlock when self crashes and calls stop() on finally
 
-    def _spawn(self, func: Callable, *args, **kwargs) -> gevent.Greenlet:
+    def _spawn_later(self, func: Callable, delay: int, *args, **kwargs) -> gevent.Greenlet:
         """ Spawn a sub-task and ensures an error on it crashes self/main greenlet """
 
         def on_success(greenlet):
             if greenlet in self.greenlets:
                 self.greenlets.remove(greenlet)
 
-        greenlet = gevent.spawn(func, *args, **kwargs)
+        greenlet = gevent.Greenlet(func, *args, **kwargs)
         greenlet.link_exception(self.on_error)
         greenlet.link_value(on_success)
         self.greenlets.append(greenlet)
+        if delay:
+            greenlet.start_later(delay)
+        else:
+            greenlet.start()
         return greenlet
 
+    def _spawn(self, func: Callable, *args, **kwargs) -> gevent.Greenlet:
+        """ Spawn a sub-task and ensures an error on it crashes self/main greenlet """
+        return self._spawn_later(func, 0, *args, **kwargs)
+
```

Comment:
I think we could have just one function with e.g. `_schedule` where if delay is not present it will default to the main loop's schedule.

Reply:
It would be nice if those utilities were part of the `Runnable` base class. But as I said above, let's do things like that separately.

Reply:
Very true. Should either me or you create issues?

Reply:
If you have time, please do it :)